kafka-rest/src/main/java/io/confluent/kafkarest/response/JsonStreamMessageBodyReader.java

```diff
@@ -53,6 +56,18 @@ public JsonStream<?> readFrom(
       MultivaluedMap<String, String> httpHeaders,
       InputStream entityStream)
       throws WebApplicationException {
+
+    // We know that the media type is 'application/json' but there might be an optional charset too
+    if (mediaType.getParameters() != null) {
+      String charset = mediaType.getParameters().get(MediaType.CHARSET_PARAMETER);
+      if (charset != null) {
+        if (!charset.equalsIgnoreCase("UTF-8") && !charset.equalsIgnoreCase("ISO-8859-1")) {
+          throw new WebApplicationException(
+              "Unsupported charset in Content-Type header.", Status.BAD_REQUEST);
```

Comment:
Should we also provide the list of valid charsets in the exception?

Reply:
That probably is a good point.